src/_pytest/assertion/util.py

```diff
@@ -169,23 +190,56 @@ def assertrepr_compare(config, op: str, left: Any, right: Any) -> Optional[List[
 
 
 def _compare_eq_any(left: Any, right: Any, verbose: int = 0) -> List[str]:
-    explanation = []  # type: List[str]
+    """
+    Compare 2 objects.
+    The generic comparison function, which calls other functions based on the input's type.
+
+    :param left
```

Comment:
Probably the empty parameter docs were added by your IDE automatically, but could you please remove the empty ones, or even better, add a description?

Also I think in general `:type` is redundant when we are already annotating the function signature (as is the case here), so they should be removed.

Reply:
Sure, I agree with both.